### ISSUE TITLE
[Book] Add custom dataset, loader and batcher detailed example

### ISSUE THREAD

**Issue by laggui** [tracel-ai/burn#1825]
For use cases that differ from the typical examples we have, implementing a custom dataset with loader and batcher seems to be a common struggle among new users.

We should provide a detailed example to help with the user implementation.

**Comment by jin-eld** [comment]
Thank you for opening an issue for this, adding some info as a follow up to our chat:

from a new user's perspective I was missing a detailed explanation on how the Dataset, DataLoader and Batcher are related to each other, what exactly each of them is responsible for and how they should be constructed and used from scratch, when trying to work with new training data. The examples that are currently provided in the book are one level beyond that, because they conveniently use already available Datasets (like MNIST), so a very basic example from scratch would be very helpful to get an understanding on how to get your custom data into Burn.
